File: 283. Move Zeroes.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        ptr = 0
        nonZeroPtr = 0
        while(ptr < len(nums)):
            if(nums[ptr] != 0):
                ptr += 1
                continue
            nonZeroPtr = ptr + 1
            while(nonZeroPtr < len(nums) and nums[nonZeroPtr] == 0):
                nonZeroPtr += 1
            if(nonZeroPtr >= len(nums)):
                break
            nums[ptr], nums[nonZeroPtr] = nums[nonZeroPtr], nums[ptr]
            ptr += 1
            # print(nums)
                
        

File: 283. Move Zeroes.java
class Solution {
    public void moveZeroes(int[] nums) {
        int ptr = 0;
        int nonZeroPtr = 0;
        for(ptr = 0; ptr < nums.length; ptr++) {
            if(nums[ptr] != 0){
                continue;
            }
            nonZeroPtr = ptr + 1;
            while(nonZeroPtr < nums.length && nums[nonZeroPtr] == 0){
                nonZeroPtr += 1;
            }
            if(nonZeroPtr >= nums.length){
                break;
            }
            int temp = nums[nonZeroPtr];
            nums[nonZeroPtr] = nums[ptr];
            nums[ptr] = temp;
        }
    }
}